Mathlib/Tactic/Linter/Lint.lean

```diff
@@ -211,6 +211,43 @@ initialize addLinter cdotLinter
 
 end CDotLinter
 
+/-!
+# The `dollarSyntax` linter
+
+The `dollarSyntax` linter flags uses of `<|` that are achieved by typing `$`.
+-/
```

Comment:
How about adding a sentence of justification, here and to the `register option`?
"These are disallowed by the mathlib style guide, as using `<|` pairs better with `|>`."